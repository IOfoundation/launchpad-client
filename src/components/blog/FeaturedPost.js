import React from 'react';
import {PropTypes} from 'prop-types';

const FeaturedPost = props => {
  const {imageSrc, title, description} = props;

  return (
    <div className="featured-post">
      <div className="featured-post__img-container">
        <img className="featured-post__img-container__img" src={imageSrc} />
      </div>
      <h3 className="featured-post__title text-bold">{title}</h3>
      <p className="featured-post__description">{description}</p>
    </div>
  );
};

FeaturedPost.propTypes = {
  description: PropTypes.string.isRequired,
  imageSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default FeaturedPost;
