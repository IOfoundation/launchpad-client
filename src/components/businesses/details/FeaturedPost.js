import React from 'react';
import {PropTypes} from 'prop-types';

export const FeaturedPost = props => {
  const {description, date} = props;

  return (
    <div className="featured-post">
      <p className="featured-post__description text-bold">{description}</p>
      <p className="featured-post__date text-thin">{date}</p>
    </div>
  );
};

FeaturedPost.propTypes = {
  date: PropTypes.string,
  description: PropTypes.string,
};

export default FeaturedPost;
