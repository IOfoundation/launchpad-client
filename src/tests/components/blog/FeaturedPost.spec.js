import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FeaturedPost from '../../../components/blog/FeaturedPost';

const props = {
  description: 'testing description',
  imageSrc: 'testimage',
  title: 'title',
};

describe('<FeaturedPost />', () => {
  it('renders snapshot of Category', () => {
    const wrapper = shallow(<FeaturedPost {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
