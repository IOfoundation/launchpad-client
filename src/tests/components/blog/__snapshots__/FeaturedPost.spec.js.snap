// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedPost /> renders snapshot of Category 1`] = `
<div
  className="featured-post"
>
  <div
    className="featured-post__img-container"
  >
    <img
      className="featured-post__img-container__img"
      src="testimage"
    />
  </div>
  <h3
    className="featured-post__title text-bold"
  >
    title
  </h3>
  <p
    className="featured-post__description"
  >
    testing description
  </p>
</div>
`;
